alica_engine/include/engine/IAlicaTrace.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <optional>
+#include <memory>
+
+namespace alica
+{
+
+class IAlicaTrace
+{
+public:
+    virtual ~IAlicaTrace() = default;
+    IAlicaTrace() = default;
```

Comment:
there is no need for constructors in an interface class. The factory implementation is responsible for creating trace objects